src/Compilers/CSharp/Portable/CommandLine/CSharpCompiler.cs

```diff
@@ -377,5 +377,35 @@ private protected override GeneratorDriver CreateGeneratorDriver(ParseOptions pa
         {
             return CSharpGeneratorDriver.Create(generators, additionalTexts, (CSharpParseOptions)parseOptions, analyzerConfigOptionsProvider);
         }
+
+        private protected override void DiagnoseBadAccesses(TextWriter consoleOutput, ErrorLogger? errorLogger, Compilation compilation, ImmutableArray<Diagnostic> diagnostics)
+        {
+            DiagnosticBag newDiagnostics = DiagnosticBag.GetInstance();
+            foreach (var diag in diagnostics)
+            {
+                var location = diag.Location;
+                var symbol = diag switch
+                {
+                    { Code: (int)ErrorCode.ERR_BadAccess, Arguments: [Symbol s] } => s,
+                    { Code: (int)ErrorCode.ERR_InaccessibleGetter, Arguments: [Symbol s] } => s,
+                    { Code: (int)ErrorCode.ERR_InaccessibleSetter, Arguments: [Symbol s] } => s,
+                    { Code: (int)ErrorCode.ERR_ImplicitImplementationOfInaccessibleInterfaceMember, Arguments: [_, Symbol s, _] } => s,
+                    _ => null
+                };
+
+                if (symbol is null || ReferenceEquals(compilation.Assembly, symbol.ContainingAssembly))
+                {
+                    // Can't be IVT related
+                    continue;
+                }
+
+                // '{0}' is defined in assembly '{1}'.
+                newDiagnostics.Add(new CSDiagnostic(
+                    new CSDiagnosticInfo(ErrorCode.ERR_SymbolDefinedInAssembly, symbol, symbol.ContainingAssembly),
```

Comment:
Why not add this information unconditionally to the existing diagnostics (improve them instead of adding a secondary diagnostic)?

Reply:
The existing diagnostics are spread out in a number of locations throughout the compiler, and I don't want to refactor them all to potentially report the existing message (in the case it can't possibly be related to an IVT) or an IVT-enhanced message. I don't think these messages are useful in most cases, only when you're trying to diagnose these issues, so I'd rather keep them separate with a post-processing step.